x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleRunner.java

```diff
@@ -65,13 +92,23 @@ public void runPolicy(String policy, IndexMetaData indexMetaData, ClusterState c
                 + "] with policy [" + policy + "] is not recognized");
             return;
         }
-        logger.debug("running policy with current-step[" + currentStep.getKey() + "]");
+        logger.debug("running policy with current-step [" + currentStep.getKey() + "]");
         if (currentStep instanceof TerminalPolicyStep) {
             logger.debug("policy [" + policy + "] for index [" + indexMetaData.getIndex().getName() + "] complete, skipping execution");
+            return;
         } else if (currentStep instanceof ErrorStep) {
             logger.debug(
-                    "policy [" + policy + "] for index [" + indexMetaData.getIndex().getName() + "] on an error step, skipping execution");
-        } else if (currentStep instanceof ClusterStateActionStep || currentStep instanceof ClusterStateWaitStep) {
+                "policy [" + policy + "] for index [" + indexMetaData.getIndex().getName() + "] on an error step, skipping execution");
+            return;
+        } else if (currentStep instanceof PhaseCompleteStep) {
+            // Only proceed to the next step if enough time has elapsed to go into the next phase
```

Comment:
We do log a message like:

```
[foo] checking for index age to be at least [30d] before performing actions in the "warm" phase. Now: <now>, lifecycle date: <date>, age: [17.1d/17230183s]
```

Every time we check whether the index is old enough (at the TRACE level), do you think that's sufficient?

Reply:
I think this may be worth bumping up to `DEBUG`? I remember it being mentioned that trace is so verbose that it slows things down. So maybe exposing it at the debug level is more sane?

Reply:
Sounds good